### DISCUSSION TITLE
Question on frequency rule behavior - need some light on how run_every and  timeframe params are used to determine a match

### DISCUSSION THREAD

**Discussion by rpotnuru** [jertel/elastalert2#554] [Q&A] [answered]
Hey Guys, 
We are trying to run few frequency type rules, and the global run_every is set to 1 min and the timeframe in my frequency rule is set to 5 mins. 
When I verify the start time and end time of the rule stats under status index, I always see that the rule is running for the only 1 min, instead of 5 mins. Could some one please help me understand the behavior of frequency rule scheduling.

Appreciate your help in this regard.

**Comment by jertel** [comment, marked as answer]
Hello. The subject of this discussion is misleading. It's stating that frequency rules do not respect the timeframe parameter. However, it appears more like you are unsure of how ElastAlert 2 is designed to function. 

All rules are queried against Elasticsearch every 1 minute, or as specified in the `run_every` parameter. Upon each execution, the rule looks backward over the specified `timeframe` to determine if an alert should be triggered.

**Reply by rbkumar88** [reply]
Hi, @jertel  but during rule execution, its not looking backward over the specified `timeframe` but rather only till the last time it ran.

**Reply by jertel** [reply]
It collects in memory what is sees every minute, and compares that total to the `num_events` parameter. See the `ruletypes.py` source code between lines 232 and 271.

https://github.com/jertel/elastalert2/blob/27deb8a61dd48798c4686ec95d3e48909903a694/elastalert/ruletypes.py#L232

ElastAlert 2 needs to run for the entire `timeframe` duration before it will start alerting.

**Reply by rbkumar88** [reply]
does it work the same way when we have use_count_query set as true

**Reply by jertel** [reply]
Yes, see lines 213-222.

**Comment by rpotnuru** [comment]
thank you @jertel  for a quick response, I updated the subject.

Does that mean that after each run, is sums up the matching events in last 5 runs (in my case)  to determine whether there is a match or not?
